services/wallet/model/model.go

```diff
@@ -1,8 +1,53 @@
 package model
 
-import "errors"
+import (
+	"encoding/base64"
+	"time"
 
-var ErrNoWalletCustodian = errors.New("model: no linked wallet custodian")
+	uuid "github.com/satori/go.uuid"
+)
+
+const (
+	ErrWalletNotWhitelisted Error = "model: wallet not whitelisted"
+	ErrNotFound             Error = "model: not found"
+	ErrChallengeExpired     Error = "model: challenge expired"
+	ErrNoRowsDeleted        Error = "model: no rows deleted"
+	ErrNotInserted          Error = "model: not inserted"
+	ErrNoWalletCustodian    Error = "model: no linked wallet custodian"
+	ErrInternalServer       Error = "model: internal server error"
+	ErrWalletNotFound       Error = "model: wallet not found"
+)
+
+type AllowListEntry struct {
+	PaymentID uuid.UUID `db:"payment_id"`
+	CreatedAt time.Time `db:"created_at"`
+}
+
+type Challenge struct {
+	ID        string    `db:"id"`
+	CreatedAt time.Time `db:"created_at"`
+	Nonce     string    `db:"nonce"`
+}
+
+func NewChallenge(id string) Challenge {
+	return Challenge{
+		ID:        id,
+		CreatedAt: time.Now(),
+		Nonce:     base64.URLEncoding.EncodeToString(uuid.NewV4().Bytes()),
+	}
+}
+
+func (c Challenge) IsValid(now time.Time) error {
```

Comment:
Even though we use it as a value, declaring methods on pointer receivers prevents unnecessary copies. It's transparent for the caller, and the compiler will insert obtaining the address for us.

```suggestion
func (c *Challenge) IsValid(now time.Time) error {
```